outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/web/OutboundDialerController.java

```diff
@@ -0,0 +1,147 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import java.util.regex.Pattern;
+
+/**
+ * OutboundDialerController
+ */
+@Controller
+public class OutboundDialerController {
+
+    public static final String NOT_PRESENT = "<%s: Not Present>";
+    public static final String INVALID = "<%s: Invalid>";
+    public static final Pattern TARGET_FILENAME_PATTERN = Pattern.compile("OBD_NMS[1-9]_20[0-9]{12}\\.csv");
+
+    @Autowired
+    private OutboundDialerService outboundDialerService;
+
+    private static boolean validateFieldPresent(StringBuilder errors, String fieldName, Object value) {
+        if (value != null) {
+            return true;
+        }
+        errors.append(String.format(NOT_PRESENT, fieldName));
+        return false;
+    }
+
+    private static boolean validateTargetFileName(StringBuilder errors, String targetFileName) {
+        if (validateFieldPresent(errors, "fileName", targetFileName)) {
+            if (TARGET_FILENAME_PATTERN.matcher(targetFileName).matches()) {
+                return true;
+            } else {
+                errors.append(String.format(INVALID, targetFileName));
+                return false;
+            }
+        }
+        return false;
+    }
+
+    private static boolean validateCdrFileInfo(StringBuilder errors, CdrFileNotificationRequestFileInfo fileInfo,
+        String fieldName, String targetFileName) {
+
+        boolean valid = true;
+
+        if (fileInfo == null) {
+            errors.append(String.format(NOT_PRESENT, fieldName));
+            return false;
+        }
+
+        if (validateFieldPresent(errors, "cdrFile", fileInfo.getCdrFile())) {
+            if (!fileInfo.getCdrFile().equals(fieldName + "_" + targetFileName)) {
+                errors.append(String.format(INVALID, fieldName));
+                valid = false;
+            }
+        }
+        
+        if (!validateFieldPresent(errors, "checksum", fileInfo.getChecksum())) {
+            valid = false;
+        }
+
+        if (fileInfo.getRecordsCount() < 0) {
+            errors.append(String.format(INVALID, "recordsCount"));
+            valid = false;
+        }
+
+        return valid;
+    }
+
+    /**
+     * 4.2.6
+     * CDR File Notification
+     *
+     * IVR shall invoke this API to notify when a CDR file is ready.
+     */
+    @RequestMapping(value = "/cdrFileNotification",
+            method = RequestMethod.POST,
+            headers = { "Content-type=application/json" })
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public void notifyNewCdrFile(@RequestBody CdrFileNotificationRequest cdrFileNotificationRequest) {
+        StringBuilder failureReasons = new StringBuilder();
+        validateTargetFileName(failureReasons, cdrFileNotificationRequest.getFileName());
```

Comment:
4.2.6.1.4 mentions the following sample body
```json
{
   "fileName": "OBD_NMS1_20150127090000.csv",
   "cdrSummary":
   {
      "cdrFile": "cdrSummary_OBD_NMS1_20150201090000.csv",
      "checksum": "xxxxxx",
      "recordsCount": 5000
   },
   "cdrDetail":
   {
      "cdrFile": "cdrDetail_OBD_NMS1_20150201090000.csv",
      "checksum" : "xxxxxx",
      "recordsCount": 9900
   }
}
```
`fileName` is what we passed to the IVR system originally, shouldn't we be validating `cdrSummary` & `cdrDetail` instead? (or in addition?)